adapters/beachfront/beachfront.go

```diff
@@ -197,289 +184,327 @@ func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *a
 	return reqs, errs
 }
 
-func preprocess(request *openrtb.BidRequest) (beachfrontReqs beachfrontRequests, errs []error) {
-	var videoImps = make([]openrtb.Imp, 0)
-	var bannerImps = make([]openrtb.Imp, 0)
+func testForValidBanner(request *openrtb.BidRequest, index int) bool {
```

Comment:
Please consider passing the imp under inspection to this function rather than passing the entire request and the index of the imp as this function requires only the imp to do what it needs to do

Reply:
I appears there is no reason for me to do any of this anyway. The banner sizes are all verified in auction.go around L441. If the sizes are not valid, the entire request would have gotten dumped. All I really need to do is check if there is a Banner. It is already known to be valid.

In my local changes I'm now just doing this:

```Go
	for i := 0; i < len(request.Imp); i++ {
		if !wantsBanner {
			wantsBanner = request.Imp[i].Banner != nil
		}
```